api/incubator/src/test/java/io/opentelemetry/api/incubator/propagation/CaseInsensitiveMapTest.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.api.incubator.propagation;
+
+import static io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatCode;
+
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.jupiter.api.Test;
+
+class CaseInsensitiveMapTest {
+
+  @Test
+  void createByConstructor() {
+    Map<String, String> map = new HashMap<>();
+    map.put("KeY1", "test");
+    map.put("KeY2", "test2");
+
+    CaseInsensitiveMap caseInsensitiveMap = new CaseInsensitiveMap(map);
+
+    Method getKeyLowerCaseMethod = null;
+    try {
+      getKeyLowerCaseMethod =
+          CaseInsensitiveMap.class.getDeclaredMethod("getKeyLowerCase", String.class);
+      getKeyLowerCaseMethod.setAccessible(true);
+    } catch (NoSuchMethodException e) {
+      throw new RuntimeException(" CaseInsensitiveMap class getKeyLowerCase method not found ", e);
+    }
```

Comment:
It would be better to just have the method be default-access, and annotate it as `@VisibleForTesting`. Going with reflection means that if someone renames the method, this test will break; if we call the method directly, automated refactoring tools will pick up the callsite and update it automatically.